src/lib/Form/Type/ContentType/FieldDefinitionsCollectionType.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form\Type\ContentType;
+
+use eZ\Publish\Core\Helper\FieldsGroups\FieldsGroupsList;
+use EzSystems\EzPlatformAdminUi\Form\Type\FieldDefinition\FieldDefinitionType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class FieldDefinitionsCollectionType extends AbstractType
+{
+    /** @var \eZ\Publish\Core\Helper\FieldsGroups\FieldsGroupsList */
+    private $fieldsGroupsList;
+
+    /**
+     * @param \eZ\Publish\Core\Helper\FieldsGroups\FieldsGroupsList $fieldsGroupsListHelper
+     */
+    public function __construct(FieldsGroupsList $fieldsGroupsListHelper)
+    {
+        $this->fieldsGroupsList = $fieldsGroupsListHelper;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        foreach ($this->fieldsGroupsList->getGroups() as $identifier => $group) {
+            if ($identifier === '') {
+                $identifier = $this->fieldsGroupsList->getDefaultGroup();
+            }
+
+            $builder->add($identifier, CollectionType::class, [
+                'entry_type' => FieldDefinitionType::class,
+                'entry_options' => [
+                    'languageCode' => $options['languageCode'],
+                    'mainLanguageCode' => $options['mainLanguageCode'],
+                ],
+                'label' => /** @Desc("Content Field definitions") */ 'content_type.field_definitions_data',
+                'allow_add' => true,
+                'allow_delete' => true,
+                'prototype' => false,
+            ]);
+        }
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setDefaults([
+                'translation_domain' => 'content_type',
+                'mainLanguageCode' => null,
```

Comment:
```suggestion
                'main_language_code' => null,
```

Reply:
:-1: because it's an option that is passed from the parent. If it weren't so, it would be ok to change it, but because of this if we were to use snake_case, we would have two different variants in use.

Reply:
Then I would just proceed and change the option name in the parent as well. It kinda looks bad if we use snake case for options, contradicts with what Symfony does.

Reply:
Should we deprecate it (with `trigger_error`) in `3.x` then, and silently convert to snake_case?

This PR targets `4.x`, so a small PR would be going first for this specifically?

Or do we ignore it, treating it as an internal thing?

Reply:
> Should we deprecate it (with `trigger_error`) in `3.x` then, and silently convert to snake_case?
> 
> This PR targets `4.x`, so a small PR would be going first for this specifically?
> 
> Or do we ignore it, treating it as an internal thing?

Depends it we documented extending and relying on those options.

Reply:
@alongosz I've looked through our docs, and any mentions of `mainLanguageCode` are in context of Content objects and their structs, not their forms. So in theory we could simply replace `mainLanguageCode` option with `main_language_code` whenever this form type is used, treating it as internal.

Is that ok with you guys? @alongosz @webhdx  @mikadamczyk ?

Reply:
> @alongosz I've looked through our docs, and any mentions of `mainLanguageCode` are in context of Content objects and their structs, not their forms. So in theory we could simply replace `mainLanguageCode` option with `main_language_code` whenever this form type is used, treating it as internal.
> 
> Is that ok with you guys? @alongosz @webhdx @mikadamczyk ?

No objections from my side, but this is mostly up to the AdminUI Maintainers ;)

Reply:
I am ok with this

Reply:
Considering how those forms are created, I'm not certain that I should go forward with this change anymore:

![image](https://user-images.githubusercontent.com/3183926/134322306-90688c39-95e1-4bab-acde-cfa1c835c27b.png)

Unfortunately that's also not the only instance that `"mainLanguageCode"` is used as an option. To maintain consistency I'd leave it as-is, unless you'd like that option with automatic conversion - although argument can be made that it should be done completely separately, and with all instances in consideration.

Therefore, out of scope?